lib/apis/v3/felixconfig.go

```diff
@@ -265,6 +265,35 @@ type FelixConfigurationSpec struct {
 	// modes can use XDP. This is not recommended since it doesn't provide better performance than
 	// iptables. [Default: false]
 	GenericXDPEnabled *bool `json:"genericXDPEnabled,omitempty" confignamev1:"GenericXDPEnabled"`
+
+	// BPFEnabled, if enabled Felix will use the BPF dataplane. [Default: false]
+	BPFEnabled *bool `json:"bpfEnabled,omitempty" validate:"omitempty,"`
+	// BPFLogLevel controls the log level of the BPF programs when in BPF dataplane mode.  One of "Off", "Info", or
+	// "Debug". [Default: Off].
+	BPFLogLevel string `json:"bpfLogLevel" validate:"omitempty,bpfLogLevel"`
+	// BPFDataIfacePattern is a regular expression that controls which interfaces Felix should attach BPF programs to
+	// in order to catch traffic to/from the network.  This needs to match the interfaces that calico workload traffic
+	// flows over as well as any interfaces that handle incoming traffic to nodeports and services from outside the
+	// cluster.  It should not match the workload interfaces (usually named cali...).  [Default: ^(en.*|eth.*|tunl0$)]
+	BPFDataIfacePattern string `json:"bpfDataIfacePattern,omitempty" validate:"omitempty,regexp"`
+	// BPFConnectTimeLoadBalancingEnabled when in BPF mode, controls whether Felix installs the connection-time load
+	// balancer.  The connect-time load balancer is required for the host to be able to reach Kubernetes services
+	// and it improves the performance of pod-to-service connections.  The only reason to disable it is for debugging
+	// purposes.  [Default: true]
+	BPFConnectTimeLoadBalancingEnabled *bool `json:"bpfConnectTimeLoadBalancingEnabled,omitempty" validate:"omitempty"`
+	// BPFExternalServiceMode in BPF mode, controls how connections from outside the cluster to services (node ports
+	// and cluster IPs) are forwarded to remote workloads.  If set to "Tunnel" then both request and response traffic
+	// is tunneled to the remote node.  If set to "DSR", the request traffic is tunneled but the response traffic
+	// is sent directly from the remote node.  In "DSR" mode, the remote node appears to use the IP of the ingress
+	// node; this requires a permissive L2 network.  [Default: Tunnel]
+	BPFExternalServiceMode string `json:"bpfExternalServiceMode,omitempty" validate:"omitempty,bpfServiceMode"`
+	// BPFKubeProxyIptablesCleanupEnabled, if enabled in BPF mode, Felix will pro-actively clean up kube-proxy's
+	// iptables chains.  Should only be enabled if kube-proxy is not running.  [Default: true]
```

Comment:
Could this be easily autodetected, and the need for a config field removed?

Reply:
(Or maybe a straightforward script outside our product?)

Reply:
Added this because it scratched an itch when we were testing.  Agree would be nice to move to auto-detection if possible but I'm not sure it is with things like hyperkube and different PID namespaces :thinking: